pkg/vcd/client/client.go

```diff
@@ -22,77 +22,56 @@ import (
 	"sync"
 
 	"github.com/vmware/cloud-provider-for-cloud-director/pkg/vcdsdk"
-	"gopkg.in/yaml.v3"
 )
 
-const (
-	config_env   = "CDCLI_CONFIG"
-	home_env     = "HOME"
-	cfg_dirname  = ".cd-cli"
-	cfg_filename = "config.yaml"
-)
-
-type Config struct {
-	RefreshToken string `yaml:"refreshToken,omitempty"`
-	Org          string `yaml:"org,omitempty"`
-	Site         string `yaml:"site"`
-	Ovdc         string `yaml:"ovdc,omitempty"`
-	Insecure     bool   `yaml:"insecure,omitempty"`
-	Username     string `yaml:"username,omitempty"`
-	Password     string `yaml:"password,omitempty"`
-}
-
 type Cache struct {
 	atStart sync.Once
 	client  *vcdsdk.Client
 }
 
-func (cache *Cache) lazyInit(items bool) {
+func (cache *Cache) lazyInit(verbose bool, context string) {
 	cache.atStart.Do(func() {
 		cfg, err := parseConfig()
 		if err != nil {
 			log.Fatal(err)
 		}
-		cache.client, err = makeClient(cfg, items)
+		cache.client, err = makeClient(cfg, verbose, context)
 		if err != nil {
 			log.Fatal(err)
 		}
 	})
 }
 
-// it assumes the config to be placed in ~/.cd-cli/config.yaml
-// or under path contained in $CDCLI_CONFIG env var
-func parseConfig() (*Config, error) {
-	var path string
-	path, found := os.LookupEnv(config_env)
-	if !found {
-		p, foundHome := os.LookupEnv(home_env)
-		path = fmt.Sprintf("%s/%s/%s", p, cfg_dirname, cfg_filename)
-		if !foundHome {
-			log.Fatal(fmt.Sprintf("Place your config to $%s/%s/%s or set %s", home_env, cfg_dirname, cfg_filename, config_env))
-		}
-	}
-	yfile, err := os.ReadFile(fmt.Sprintf("%s", path))
-	if err != nil {
-		log.Fatal(fmt.Errorf("Unable to open configuration file, make sure it exist (~/.cd-cli/config.yaml)\n%w", err))
+func makeClient(allContexts *Config, verbose bool, context string) (*vcdsdk.Client, error) {
+	var cfg *VCDInfo
+	currentContext := context
+	if len(currentContext) == 0 {
+		currentContext = allContexts.CurrentContext
 	}
-	data := &Config{}
-	err2 := yaml.Unmarshal(yfile, &data)
-
-	if err2 != nil {
-		log.Fatal(err2)
+	if len(currentContext) == 0 {
+		// current context is not specified, take the first occurende in the cfg file
+		if len(allContexts.Contexts) > 0 {
```

Comment:
I think it is better to use the first context as default when there is only one context.

Reply:
that's how this actually works: `cfg = &allContexts.Contexts[0]`

there was a typo in the comment and I added also the `currentContext = cfg.Name` under this if so that line 75 will print it if `verbose==true`

Reply:
Personally, I would just give an error if there are multiple contexts and there is no explicit currentContext but it is up to you. Approving the PR.